Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+        Divide(LHSWorkspace=ws, RHSWorkspace=beam_ws, OutputWorkspace=ws)
+        return ws
+
+    def _subtract_background(self, ws, container_ws, transmission_ws):
+        """Subtracts empty container and cadmium scaled by transmission."""
+        if self._mode != 'TOF':
+            absorber_ws = self.getPropertyValue('AbsorberInputWorkspace')
+            if absorber_ws == "":
+                raise RuntimeError("Absorber input workspace needs to be provided for non-TOF background subtraction.")
+        for entry_no, entry in enumerate(mtd[ws]):
+            container_entry = mtd[container_ws].getItem(entry_no).name()
+            mtd[container_entry].setYUnit('Counts/Counts')
+            mtd[transmission_ws].setYUnit('Counts/Counts')
+            if self._mode != 'TOF':
+                absorber_entry = mtd[absorber_ws].getItem(entry_no).name()
+                mtd[absorber_entry].setYUnit('Counts/Counts')
+                Minus(LHSWorkspace=entry,
+                      RHSWorkspace=mtd[transmission_ws] * mtd[container_entry] + (1-mtd[transmission_ws]) * mtd[absorber_entry],
+                      OutputWorkspace=entry)
+            else:
+                raise RuntimeError("TOF requires elastic channel definition")
+        return ws
+
+    def _calculate_polarizing_efficiencies(self, ws):
+        """Calculates the polarizing efficiencies using quartz data."""
+        flipper_eff = 1.0 # this could be extracted from data if 4 measurements are done
+        nMeasurementsPerPOL = 2
+        tmp_names = []
+        index = 0
+        if self.getProperty('AverageScan').value:
+            ws = self._merge_polarisations(ws, average_detectors=True)
+        for entry_no in range(1, mtd[ws].getNumberOfEntries()+1, nMeasurementsPerPOL):
+            # two polarizer-analyzer states, fixed flipper_eff
+            ws_00 = mtd[ws].getItem(entry_no).name()
+            ws_01 = mtd[ws].getItem(entry_no-1).name()
+            tmp_name = '{0}_{1}_{2}'.format(ws[2:], mtd[ws_00].getRun().getLogData('POL.actual_state').value, index)
+            Divide(LHSWorkspace=mtd[ws_00]-mtd[ws_01],
+                   RHSWorkspace=(2*flipper_eff-1)*mtd[ws_00]+mtd[ws_01],
+                   OutputWorkspace=tmp_name)
+            tmp_names.append(tmp_name)
+            if self._method_data_structure == 'Uniaxial' and entry_no % 2 == 1:
+                index += 1
+            elif self._method_data_structure == 'XYZ' and entry_no % 6 == 5:
+                index += 1
+            elif self._method_data_structure == '10p' and entry_no % 10 == 9:
+                index += 1
+        GroupWorkspaces(InputWorkspaces=tmp_names, OutputWorkspace='tmp')
+        DeleteWorkspaces(ws)
+        RenameWorkspace(InputWorkspace='tmp', OutputWorkspace=ws)
+        return ws
+
+    def _detector_analyser_energy_efficiency(self, ws):
+        """Corrects for the detector and analyser energy efficiency."""
+        correction_formula = "1.0 - exp(-13.153/sqrt(e))"
```

Comment:
This formula should go to instrument parameters file as a mu parser expression. You can check how it's done in IN4.